### PR TITLE
OS-8510 Update OpenSSH to 9.6p1

### PR DESCRIPTION
**Pull request by danmcd**
Subject says it all.

Needed to fix two patches, first because of context-change, second because of new OpenSSL-dependent ifdefs that need the SUNW prefix.

### CODE REVIEW

File: openssh/Patches/0008-Add-DisableBanner-option.patch [hunk 1]
```diff
@@ -101,10 +101,10 @@ index 2ce1b4c3..0ca0ebe0 100644
  }       Options;
  
  #define SSH_PUBKEY_AUTH_NO	0x00
-@@ -224,6 +227,13 @@ typedef struct {
+@@ -234,6 +237,13 @@ typedef struct {
  const char *kex_default_pk_alg(void);
  char	*ssh_connection_hash(const char *thishost, const char *host,
-     const char *portstr, const char *user);
+     const char *portstr, const char *user, const char *jump_host);
```
Comment (danmcd) [inline openssh/Patches/0008-Add-DisableBanner-option.patch:107]: This was the problem-child context.

File: openssh/Patches/1001-sunw_ssl.patch [hunk 2]
```diff
@@ -278,6 +280,212 @@ index f6796b3b..9a333d64 100644
...
+diff --git a/ssh-pkcs11-client.c b/ssh-pkcs11-client.c
+index 82e86a518..81db1a622 100644
+--- a/ssh-pkcs11-client.c
++++ b/ssh-pkcs11-client.c
+@@ -48,7 +48,7 @@
+ 
+ #include "openbsd-compat/openssl-compat.h"
+ 
+-#if !defined(OPENSSL_HAS_ECC) || !defined(HAVE_EC_KEY_METHOD_NEW)
++#if !defined(OPENSSL_HAS_ECC) || !defined(HAVE_SUNW_EC_KEY_METHOD_NEW)
```
Comment (danmcd) [inline openssh/Patches/1001-sunw_ssl.patch:292]: Needed new SUNW prefixes here.

**Review by danmcd — COMMENTED** [review]
Background indicator comments...

**Review by travispaul — APPROVED** [review]

**Review by bahamat — APPROVED** [review]
This looks good to me. If we have a confirmed build, I'll IA.

**Pull request merged**